modules/getweblinks.py

```diff
@@ -1,7 +1,58 @@
+import requests
+
 from modules.net_utils import get_urls_from_page, get_url_status
-from modules import pagereader
 from bs4 import BeautifulSoup
 from modules.bcolors import Bcolors
+from threading import Thread
+from queue import Queue
+
+
+def traverse_links(links, ext, depth=0, stop_depth=None, targetLink=None):
+    """
+        Traverses links passed using Breadth First Search. You can specify stop depth
+        or specify a target to look for. The depth argument is used for recursion
+
+        Args:
+            links (list): list of urls to traverse
+            ext (string): string representing extension to use for URLs
+            depth (int): used for recursion
+            stop_depth (int): stops traversing at this depth if specified
+            targetLink (string): stops at this link if specified
+
+        Returns:
+            depth (int): depth stopped at
+    """
+
+    if depth == stop_depth:
+        return depth
+
+    toVisit = list()
+    for link in links:
+        if targetLink == link and targetLink:
```

Comment:
I want to make sure `targetLink` since Python is a dynamic language, it's impossible to tell ahead of time what items a list may contain. If a `None` were to somehow get inserted, I don't want it to return a false positive.